src/test/plugins/mvcc_delete_plugin_system_test.cpp

```diff
@@ -0,0 +1,130 @@
+
+    StorageManager::get().add_table("mvcc_test", table);
+  }  // managed by each test individually
+
+  void TearDown() override {
+    StorageManager::get().drop_table("mvcc_test");
+    StorageManager::reset();
+  }
+
+ protected:
+  // Update all values in the table to create invalidated rows.
+  void update_table() {
+    auto column = expression_functional::pqp_column_(ColumnID{0}, DataType::Int, false, "number");
+
+    auto& tm = TransactionManager::get();
+
+    const auto transaction_context = tm.new_transaction_context();
+    const auto value = static_cast<int>(_counter++ % (MAX_CHUNK_SIZE));
+    const auto expr = expression_functional::equals_(column, value);
+
+    const auto gt = std::make_shared<GetTable>("mvcc_test");
+    gt->set_transaction_context(transaction_context);
+
+    const auto validate = std::make_shared<Validate>(gt);
+    validate->set_transaction_context(transaction_context);
+
+    const auto where = std::make_shared<TableScan>(validate, expr);
+    where->set_transaction_context(transaction_context);
+
+    const auto update = std::make_shared<Update>("mvcc_test", where, where);
+    update->set_transaction_context(transaction_context);
+
+    gt->execute();
+    validate->execute();
+    where->execute();
+    update->execute();
+
+    if (!update->execute_failed()) {
+      transaction_context->commit();
+    } else {
+      transaction_context->rollback();
+      _counter--;
+    }
+  }
+
+  constexpr static size_t MAX_CHUNK_SIZE = 200;
+
+  constexpr static size_t PHYSICALLY_DELETED_CHUNKS_COUNT = 3;
+  constexpr static size_t MAX_UPDATES = MAX_CHUNK_SIZE * 1000;
+  constexpr static size_t LOOP_COUNT = 10;
+
+  size_t _deleted_chunks = 0;
+  uint64_t _counter = 0;
+};
+
+/**
+ * This test checks the number of physically deleted chunks, which means
+ * nullptrs in the table.
+ * These nullptrs are created when the plugin successfully deleted a chunk.
+ * This test was designed for _IDLE_DELAY_LOGICAL_DELETE = _IDLE_DELAY_PHYSICAL_DELETE = 1000
+ * and may fail if these constants are changed in the MvccDeletePlugin.
+ */
+TEST_F(MvccDeletePluginSystemTest, CheckPlugin) {
+  auto& pm = PluginManager::get();
+  pm.load_plugin(build_dylib_path("libMvccDeletePlugin"));
+
+  // This thread calls update_table() continuously.
+  std::unique_ptr<PausableLoopThread> table_update_thread =
+      std::make_unique<PausableLoopThread>(std::chrono::milliseconds(0), [&](size_t) { update_table(); });
+
+  for (size_t loop_count = 0;
+       _deleted_chunks < PHYSICALLY_DELETED_CHUNKS_COUNT && (_counter < MAX_UPDATES || loop_count < LOOP_COUNT);
+       ++loop_count) {
+    const auto table = StorageManager::get().get_table("mvcc_test");
+    _deleted_chunks = 0;
+
+    for (auto chunk_id = ChunkID{0}; chunk_id < table->chunk_count(); ++chunk_id) {
+      if (!table->get_chunk(chunk_id)) {
+        _deleted_chunks++;
+      }
+    }
+    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
```

Comment:
The thread is not guaranteed to be executed exactly after 1s. I'd at least wait for 50% longer.